src/main/java/com/company/elixr/springbootapplicationwithdocker/service/FileOperationServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+public class FileOperationServiceImpl implements FileOperationService {
+
+    private final FileOperationRepository fileOperationRepository;
+    private final FileInfo fileInfo;
+    @Value(Constants.LOCAL_STORAGE_FOLDER_PATH)
+    private String dirLocation;
+
+    @Override
+    public void init() {
+        try {
+            Files.createDirectories(Path.of(dirLocation));
+        } catch (Exception ex) {
+            throw new FileStorageException(Constants.ERROR_CREATING_UPLOAD_DIRECTORY);
+        }
+    }
+
+    @Override
+    public ResponseEntity<SuccessResponse> saveFile(MultipartFile file, String userName) {
+        if (Objects.equals(file.getContentType(), Constants.REQUIRED_FILE_TYPE)) {
+            try {
+                String fileName = StringUtils.cleanPath(Objects.requireNonNull(file.getOriginalFilename()));
+                fileInfo.setFileName(fileName);
+                fileInfo.setUserName(userName);
+                UUID id = fileOperationRepository.save(fileInfo).getId();
+                Path resolvedPath = Path.of(dirLocation).resolve(String.valueOf(id));
+                Files.copy(file.getInputStream(), resolvedPath, StandardCopyOption.REPLACE_EXISTING);
+                fileOperationRepository.save(fileInfo);
+                return ResponseEntity.status(HttpStatus.OK).body(SuccessResponse.builder().status(Constants.STATUS).id(id).build());
+            } catch (IOException | NoSuchElementException e) {
+                throw new FileStorageException(Constants.ERROR_UPLOADING_FILE);
+            }
+        } else {
+            throw new BadRequestException(Constants.ERROR_BAD_REQUEST_FILE_NOT_PRESENT_OR_INVALID_FILE_TYPE);
+        }
+    }
+
+    @Override
+    public ResponseEntity<SuccessResponseForGetById> findFileById(String id) {
+
+        if ((id.matches(Constants.UUID_FORMAT))) {
+            UUID uuid = UUID.fromString(id);
+            FileInfo targetFileInfo = fileOperationRepository.findById(String.valueOf(uuid)).orElseThrow(() -> new NotFoundException(Constants.ERROR_NOT_FOUND));
+            Path targetPath = Path.of(dirLocation).resolve(String.valueOf(uuid)).normalize();
+            try {
+                Resource resource = new UrlResource(targetPath.toUri());
+                if (resource.exists() || resource.isReadable()) {
+                    StringBuilder stringBuilder = new StringBuilder();
+                    BufferedReader br = new BufferedReader(new InputStreamReader(resource.getInputStream()));
+                    String line;
+                    while ((line = br.readLine()) != null) {
+                        stringBuilder.append(line);
+                    }
+                    return ResponseEntity.status(HttpStatus.OK).body(new SuccessResponseForGetById(Constants.SUCCESS, SuccessResponse.builder().userName(targetFileInfo.getUserName()).fileName(targetFileInfo.getFileName()).timeOfUpload(targetFileInfo.getTimeOfUpload()).content(stringBuilder.toString()).build()));
```

Comment:
Line is too long, split into multiple lines

Reply:
Split it into multiple lines